spyder/plugins/remoteclient/widgets/container.py

```diff
@@ -262,6 +264,30 @@ def on_kernel_started(self, ipyclient, kernel_info):
             # Connect client to the kernel
             ipyclient.connect_kernel(kernel_handler)
 
+    def on_server_version_mismatch(self, config_id, version: str):
+        """
+        Actions to take when there's a mismatch between the
+        spyder-remote-services version installed in the server and the one
+        supported by Spyder.
+        """
+        auth_method = self.get_conf(f"{config_id}/auth_method")
+        server_name = self.get_conf(f"{config_id}/{auth_method}/name")
+
+        QMessageBox.critical(
+            self,
+            _("Remote server error"),
+            _(
+                "When trying to connect to the server <b>{}</b>, we detected "
+                "that the version of <tt>spyder-remote-services</tt>, the "
+                "package Spyder needs to communicate to, is <b>{}</b>, which "
+                "is greater than the maximum one currently supported "
+                "(<b>{}</b>)."
+                "<br><br>"
+                "Please update Spyder to fix this problem."
+            ).format(server_name, version, SPYDER_REMOTE_MAX_VERSION),
```

Comment:
```suggestion
            ).format(
                server=server_name,
                srs_version=version,
                max_version=SPYDER_REMOTE_MAX_VERSION,
            ),
```

Also used named parameters for clarity, flexibility and less change of bugs and mistakes.